src/modules/olMap/services/RtVectorLayerService.js

```diff
@@ -0,0 +1,188 @@
+import { containsExtent, getCenter } from '../../../../node_modules/ol/extent';
+
+export const WebSocket_Message_Keep_Alive = 'keep-alive';
+export const WebSocket_Ports = [80, 443];
+
+/**
+ * Checks, whether a successor port is in the specified array of unique ports available or not.
+ * @param {Array<number>} ports
+ * @param {number} predecessorPort
+ * @returns {boolean}
+ */
+export const isNextPortAvailable = (ports, predecessorPort) => (ports ? ports.indexOf(predecessorPort) < ports.length - 1 : false);
+
+/**
+ * Returns the next port from the specified array of unique ports.
+ * @param {Array<number>} ports
+ * @param {number} current
+ * @returns {number}
+ */
+export const getNextPort = (ports, current = 80) => {
+	const nextPort = ports[ports.indexOf(current) + 1];
+	return nextPort === 80 ? undefined : nextPort;
+};
+
+/**
+ * Service that creates an ol layer from a {@link RtVectorGeoResource} (Websocket)
+ * and applies specific stylings if required.
+ * @class
+ * @author thiloSchlemmer
+ */
+export class RtVectorLayerService {
+	_addPortToUrl(url, port) {
+		// todo: possible refactoring to a general network util method (UrlService)
+		const pathArray = url.split('.');
+		const applyOnLastElement = (pathElements, port) => {
+			const lastElement = pathElements[pathElements.length - 1];
+			if (lastElement.indexOf('/') !== -1) {
+				const lastElementSegments = lastElement.split('/');
+				lastElementSegments[0] = lastElementSegments[0] + ':' + port;
+				return lastElementSegments.join('/');
+			} else {
+				return pathArray[pathArray.length - 1] + ':' + port;
+			}
+		};
+		pathArray[pathArray.length - 1] = applyOnLastElement(pathArray, port);
+		return pathArray.join('.');
+	}
+
+	_getFeatureReader(rtVectorGeoResource) {
+		const { MapService: mapService } = $injector.inject('MapService');
+		const destinationSrid = mapService.getSrid();
+
+		const format = mapVectorSourceTypeToFormat(rtVectorGeoResource.sourceType);
+
+		// only eWKT needs a 'per-data'-check for the srid. All other (currently) supported
+		// formats have 4326 as srid per definition
+		return rtVectorGeoResource.sourceType === VectorSourceType.EWKT
+			? (data) => {
+					const eWkt = parse(data);
+					return format
+						.readFeatures(eWkt.wkt)
+						.filter((f) => !!f.getGeometry())
+						.map((f) => {
+							f.getGeometry().transform('EPSG:' + eWkt.srid, 'EPSG:' + destinationSrid);
+							return f;
+						});
+				}
+			: (data) => {
+					return format
+						.readFeatures(data)
+						.filter((f) => !!f.getGeometry())
+						.map((f) => {
+							f.getGeometry().transform('EPSG:4326', 'EPSG:' + destinationSrid);
+							return f;
+						});
+				};
+	}
+
+	_processMessage(messageData, olVectorLayer, featureReader) {
+		const olVectorSource = olVectorLayer.getSource();
+		olVectorSource.clear();
+		const features = featureReader(messageData);
+		olVectorSource.addFeatures(features);
+	}
+
+	_fitViewOptionally(olVectorLayer, olMap, firstFit) {
```

Comment:
Maybe we should add an explanation about our fit-to-extend/change-center approach